### ISSUE TITLE
Error in running Marian with a config file

### ISSUE THREAD

**Issue by PersianNLPer** [marian-nmt/marian-dev#449]
Hi,
I am trying to run Marian with the parameters being sent in a config file (using --config option) but I get the `Aborted (core dumped)` error.
The strange part is that if I pass the options as a part of the command for calling Marian (similar to the Marian-examples) it works without any problem.
So, this means that the config parser has some issues and breaks while parsing the options.

Here is the part of the log where I get the error:

`[2019-05-24 11:44:09] [config] Model is being created with Marian v1.7.8 b13ee2c9 2019-05-10 12:16:26 +0200`
`[2019-05-24 11:44:09] Using synchronous training`
`[2019-05-24 11:44:09] [data] Loading vocabulary from JSON/Yaml file /home/amin/projects/wmt/vocab.ende.yml`
`[2019-05-24 11:44:09] [data] Setting vocabulary size for input 0 to 36000`
`[2019-05-24 11:44:09] [data] Loading vocabulary from JSON/Yaml file /home/amin/projects/wmt/vocab.ende.yml`
`[2019-05-24 11:44:10] [data] Setting vocabulary size for input 1 to 36000`
`[2019-05-24 11:44:10] Compiled without MPI support. Falling back to FakeMPIWrapper`
`[2019-05-24 11:44:10] Error: Unhandled exception of type` 'N4YAML18TypedBadConversionIfEE': yaml-cpp: error at line 1, column 1: bad conversion
`[2019-05-24 11:44:10] Error: Aborted from void unhandledException() in /home/amin/NLP/tools/NMT/marian-dev/src/common/logging.cpp:107`

`[CALL STACK]`
`[0x561a51611326]                                                       + 0x16e326`
`[0x7f0f135cfab6]                                                       + 0x92ab6`
`[0x7f0f135cfaf1]                                                       + 0x92af1`
`[0x7f0f135cfd24]                                                       + 0x92d24`
`[0x561a515374b8]                                                       + 0x944b8`
`[0x561a515da9bb]                                                       + 0x1379bb`
`[0x561a5187f269]                                                       + 0x3dc269`
`[0x561a5159b339]                                                       + 0xf8339`
`[0x561a515f56a2]                                                       + 0x1526a2`
`[0x561a51525002]                                                       + 0x82002`
`[0x561a51502645]                                                       + 0x5f645`
`[0x7f0f1298db97]    __libc_start_main                                  + 0xe7`
`[0x561a5152369a]                                                       + 0x8069a`

`Aborted (core dumped)`

**Comment by frankseide** [comment]
Which parameter and what was its value?

Any chance you can recompile Marian with debug symbols enabled?

(@Marcin, do we have debug symbols enabled by default for Release builds?)

**Comment by emjotde** [comment]
Closing due to inactivity.

**Comment by stribizhev** [comment]
@emjotde I got the same issue. It happened when I tried to re-train a model by adding a bit more sentence pairs to the training set and launched training with the previous configuration file and increased `epochs` number. I think it is related to the fact that the original model was trained on 8 GPUs, and this training was launched on just 2 GPUs. Not sure though, I will update once I find out the root cause.
